packages/cli/src/lib/server.ts

```
@@ -0,0 +1,105 @@
+import express from 'express'
+import http from 'http'
+import { once } from 'lodash'
+import logger from './logger'
+import path from 'path'
+import { loadDestination } from './destinations'
+import { Destination, CloudDestinationDefinition } from '@segment/actions-core'
+import asyncHandler from './async-handler'
+import getExchanges from './http'
+import ora from 'ora'
+import chalk from 'chalk'
+
+const app = express()
+app.use(express.json())
+
+const spinner: ora.Ora = ora()
+
+const port = process.env.PORT || 3000
+const server = http.createServer(app)
+
+app.post(
+  '/:action',
+  asyncHandler(async (req: express.Request, res: express.Response) => {
+    spinner.start(chalk`Handling ${process.env.DESTINATION} action request`)
+
+    const directory = process.env.DIRECTORY as string
+    const destinationPath = process.env.DESTINATION as string
+
+    // For now, include the slug in the path, but when we support external repos, we'll have to change this
+    const relativePath = path.join(directory, destinationPath, 'index.ts')
+    const targetDirectory = path.join(process.cwd(), relativePath)
+
+    try {
+      const def = await loadDestination(targetDirectory) // loadDestinationDefinition
+      const destination = new Destination(def as CloudDestinationDefinition)
+
+      const actionSlug = req.params.action
+
+      const action = destination.actions[actionSlug]
+
+      if (action) {
```

Should throw an error if the action was not defined or else we just keep showing the handling message to the user (`Handling google-analytics-4 action request`) giving them the impression the operation is still being processed.